### ISSUE TITLE
Distribution shows strange behavior

### ISSUE THREAD

**Issue by artemyk** [dit/dit#156]
It seems to me that multivariate distributions behave in a confusing, and possibly buggy manner.  Consider the following:
```
import dit
from dit.pid.distributions import bivariates
p = bivariates['and']

print(p['000'])  # prints "0.25"
print(p['0','0','0']) # print "0", without raising exception
print(p['0','0','A']) # raises exception, as expected
```
The issue seems to be that `p.has_outcome` thinks that `('0','0','0')` is in the sample space because `(('0','0','0') in p._sample_space)` returns `True`, which in turn happens (I think) because `CartesianProduct` treats `('0','0','0')` the same as `'000'`. However, `__getitem__` in `Distribution` can't find the probability for `('0','0','0')` because for it `('0','0','0')` is not the same as `'000'`.

**Comment by Autoplectic** [comment]
Hi Artemy,

I'll make __getitem__ more robust so that p['0', '0', '0'] returns 0.25 in that example. That would make a lot more sense. It might take me a bit, since I didn't author and don't have too much familiarity with all the sample space and lookup methods.

**Comment by Autoplectic** [comment]
ok, i think this is fixed in 64b4bcd899c36dabe467efbd9357edf3e583b50b

If you encounter any issues, please comment further and i'll try harder.
